### ISSUE TITLE
CPU memory grows up while using CUDA shared memory

### ISSUE THREAD

**Issue by Prasanna19971124** [NVIDIA/tensorrt-inference-server#1001]
**Description**
I have been working with TRTIS CUDA shared memory, I am testing YOLOv3 model with three outputs. The client allocates a shared memory during class initialization and uses that memory to read and write tensors. I noticed that cuda shared memory results in increasingly growing main memory, maybe there is some pointer which is not being freed or a new chunk of memory is being allocated every time. 

Here is the code snippet:
```python
class TensorRT_SHM_Client :

  #parameters = dict, Inferecnce parameters
  #shm_parameters = dict, GPU Related shared memory parameters

  def __init__(self, parameters = {}, shm_parameters = {}, batch_size = 15):
    self.parameters = parameters
    model_name = parameters['modelname']
    protocol = ProtocolType.from_str(parameters['protocol'])
    self.healthCTX = ServerHealthContext(parameters['url'], protocol)
    if not self.healthCTX.is_live() :
      print('Health status :', "FAILED")
    status_ctx = ServerStatusContext(parameters['url'], protocol, model_name)
    print('Server status : ', status_ctx.get_server_status())

    self.shm_parameters = shm_parameters
    self.batch_size = batch_size

    #self.labels = load_labels()
    #
    self.inference_ctx = InferContext(parameters['url'], protocol, model_name, parameters["model_version"])
    self.shared_memory_ctx = SharedMemoryControlContext(parameters['url'], protocol)

    self.create_shared_memory(self.batch_size, 416, 416, 3, np.dtype('float32'))


  def create_shared_memory(self, batch_size, w, h, c, dtype):

    print(self.parameters)

    shm_size = batch_size * w * h * c * dtype.itemsize
    shm_op_size = batch_size * 255 * 13 * 13 * dtype.itemsize
    shm_op1_size = batch_size * 255 * 26 * 26 * dtype.itemsize
    shm_op2_size = batch_size * 255 * 52 * 52 * dtype.itemsize

    print('Allocating shared memory with size : ', shm_size)

    self.shm_handle = cudashm.create_shared_memory_region("resnet_50_ip", shm_size, 0)
    self.shm_handle_op = cudashm.create_shared_memory_region("resnet_50_op", shm_op_size, 0)
    self.shm_handle_op1 = cudashm.create_shared_memory_region("resnet_50_op1", shm_op1_size, 0)
    self.shm_handle_op2 = cudashm.create_shared_memory_region("resnet_50_op2", shm_op2_size, 0)
    #print(self.shm_handle)


    self.shared_memory_ctx.cuda_register(self.shm_handle_op)
    self.shared_memory_ctx.cuda_register(self.shm_handle_op1)
    self.shared_memory_ctx.cuda_register(self.shm_handle_op2)

    self.shared_memory_ctx.cuda_register(self.shm_handle)
    


  # Call this method from a SIGINT signal handler.
  def free_shared_memory(self):
    
    self.shared_memory_ctx.unregister_all()
    cudashm.destroy_shared_memory_region(self.shm_handle)
    cudashm.destroy_shared_memory_region(self.shm_handle_op)
    cudashm.destroy_shared_memory_region(self.shm_handle_op1)
    cudashm.destroy_shared_memory_region(self.shm_handle_op2)
    print('CUDA Shared memories destroyed')


  def run_inference(self, image_batch, batch_size = 15):

    #print(self.shm_handle)

    #self.create_shared_memory_re

    #self.create_shared_memory(self.batch_size, 416, 416, 3, np.dtype('float32'))

    cudashm.set_shared_memory_region(self.shm_handle, image_batch)

    self.inference_ctx.run({
      self.parameters['input'] : (self.shm_handle, (3, 416, 416))
      }, {
        '082_convolutional'  : (InferContext.ResultFormat.RAW, self.shm_handle_op),
        '094_convolutional'  : (InferContext.ResultFormat.RAW, self.shm_handle_op1),
        '106_convolutional' : (InferContext.ResultFormat.RAW, self.shm_handle_op2)
      },
      self.batch_size
    )

```

I run this in an infinite loop where I get images from a source and feed them as batches. When I used `top` to benchmark CPU usage, I observed CPU memory growing by 4.2% of 32GB at every interval, just after 10 mins, my PC crashed and killed the process forcefully to save memory. I tried doing several hacks to fix the issue, like I tried deleting `del` the result but it didn't work out. When i used shared memory only for input I did not observe this memory growth. So I feel there is some issue at the output end of shared memory.

**TRTIS Information**
Latest version
Container pulled from NGC

**To Reproduce**
Use the above code snippet and call `run_inference()` in a while loop with some image input source and observe the  main memory usage from `top` command:

**Expected behavior**
Constant memory usage

**Comment by Prasanna19971124** [comment]
Can you help me out  with this? @GuanLuo @dougn @swarren @lukeyeager

**Comment by CoderHam** [comment]
I'm looking into this right now.

**Comment by Prasanna19971124** [comment]
Any updates on this ?

**Comment by CoderHam** [comment]
The fix #1004 was just merged into master.

**Comment by Prasanna19971124** [comment]
Thank you so much for your time ! :)
